### PR TITLE
Range Selection: Simplify range selection in redux state

### PR DESCRIPTION
**Pull request by rileyajones**
* Motivation for features / changes
The redux state currently has two values used to represent whether range selection is enabled (`stepSelectorRangeEnabled` and `linkedTimeRangeEnabled`). Now that the UX around enabling range selection is being changed to simply be a checkbox in the settings panel this doesn't make a lot of sense.

This will enable #5956 will flush out the functionality for range selection without linked time enabled.

* Technical description of changes
replace `stepSelectorRangeEnabled` and `linkedTimeRangeEnabled` with `rangeSelectionEnabled` in redux state.

* Screenshots of UI changes
N/A

* Detailed steps to verify changes work correctly (as executed by you)
1) Start tensorboard
2) Go to http://localhost:6006?allowRangeSelection&enableDataTable&enableLinkedTime
3) Enable "Link by Step" and add an end value
4) Assert that you now have two fobs (do not expect range selection to be fully functional in without "Link By Step" enabled the rest of the functionality is coming in #5956)
image
5) Assert that "Enable Range Selection" is now enabled
6) Disable "Enable Range Selection"
7) Assert that the end fob disappears
8) Re-enable "Enable Range Selection"
9) Assert the end fob reappears.

* Alternate designs / implementations considered
Do not replace the two keys and just modify the reducers and selectors to check both, this seems much messier.

### CODE REVIEW

File: tensorboard/webapp/metrics/store/metrics_reducers.ts [hunk 1]
```diff
@@ -979,18 +978,22 @@ const reducer = createReducer(
...
+    let linkedTimeSelection = state.linkedTimeSelection
+      ? {...state.linkedTimeSelection}
+      : null;
+    if (nextRangeSelectionEnabled) {
+      nextStepSelectorEnabled = nextRangeSelectionEnabled;
     }
 
     return {
       ...state,
+      linkedTimeSelection,
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/store/metrics_reducers.ts:994]: I do not understand what you are trying to accomplish with by setting this here.

File: tensorboard/webapp/metrics/store/metrics_selectors.ts [hunk 2]
```diff
@@ -371,14 +371,14 @@ export const getMetricsStepSelectorEnabled = createSelector(
 export const getMetricsStepSelectorRangeEnabled = createSelector(
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/store/metrics_selectors.ts:371]: I think we should combine getMetricsStepSelectorRangeEnabled and getMetricsLinkedTimeRangeEnabled into getMetricsRangeEnabled. Why do we want two selectors which return the same value?

File: tensorboard/webapp/metrics/store/metrics_reducers.ts (hunk 1 above)
Comment (rileyajones) [reply tensorboard/webapp/metrics/store/metrics_reducers.ts]: Looks like I missed pulling this when I split the step selector portion of out this pr.

File: tensorboard/webapp/metrics/store/metrics_selectors.ts (hunk 2 above)
Comment (rileyajones) [reply tensorboard/webapp/metrics/store/metrics_selectors.ts]: I definitely should have done that initially, done now.

**Review by JamesHollyer — APPROVED** [review]

**Pull request merged**